bokehjs/src/lib/models/plots/plot.ts

```diff
@@ -219,6 +219,7 @@ export class Plot extends LayoutDOM {
   add_layout(renderer: Annotation | GuideRenderer, side: Place = "center"): void {
     const side_renderers = this.getv(side)
     side_renderers.push(renderer as any /* XXX */)
+    this.properties[side].change.emit()
```

Comment:
This doesn't fulfill the protocol, because you have to also emit `this.change`. It would be better to change this to:
```ts
const side_renderers = this.getv(side)
this.setv({[side]: [...side_renderers, renderer as any]})
```

Reply:
What protocol? Is there any documentation about it?

Unless I misunderstood, someplace else @bryevdv has mentioned that changing a property's value without reassigning the property is expected to not trigger the `change` even on the whole model.

Reply:
> What protocol? Is there any documentation about it?

"Protocol" per the implementation (of `HasProps._setv()`). You have to emit both `change` slots. Given how much confusion this dichotomy has caused over the years, I think I will remove one for 3.0.

> (...) changing a property's value without reassigning the property is expected to not trigger the change even on the whole model.

True, but in this case this is a reassignment. `obj.property = value` is syntactic sugar for `obj.setv({property: value})`.

Reply:
I'm afraid I still don't have a complete understanding.
Right now, there are two different things:
1. Changing a property of a model: `m.x = 1`, which triggers `m.change`
2. Changing some inner state of a property of a model: `m.x.push(1)`, `m.x.y = 1`, etc, which triggers `m.x.change`.

I can see how `m.x.push()` not triggering `m.change` can be confusing and I think that triggering `m.change` (automatically!) in this case would be a good thing. Given that there's no automatic way to do so right now (although I imagine nothing precludes us from implementing it in much the same way Python does it), I agree with your initial comment. So sure, I'll make the change.

However, I don't follow what you want to remove in 3.0. I **definitely** want to be able to listen to `m.change`, and I just as definitely want to be able to listen **just** to `m.x.change`.

Reply:
> Changing a property of a model: m.x = 1, which triggers m.change

`m.x = 1` or `m.setv({x: 1})` will trigger both `m.change` and `m.x.change`.

> Changing some inner state of a property of a model: m.x.push(1), m.x.y = 1, etc, which triggers m.x.change.

This will trigger neither. When we drop IE, then we can consider using `Proxy` to monitor in-place changes. Though that will require detailed performance evaluation to check if this is actually feasible.

>  I definitely want to be able to listen to m.change, and I just as definitely want to be able to listen just to m.x.change.

I don't have a plan yet, but I would like coalesce both into one, so that you wouldn't be able to end up listening to both and duplicating work, or not emitting one and user's code silently doing nothing at all.

Reply:
> This will trigger neither

I should've been more clear, sorry. I meant that `m.x` is a proper `Model` by itself. That's why I used `m.x.change` and not `m.properties.x.change`. And that's why I expect `m.x = 1` to trigger `m.properties.x.change` and `m.change` but not `m.x.change`, although this particular expectation may be ill-placed.